onnxruntime/core/framework/allocatormgr.h

```diff
@@ -33,4 +39,24 @@ struct AllocatorCreationInfo {
 // Valid values can be found in onnxruntime_c_api.h.
 AllocatorPtr CreateAllocator(const AllocatorCreationInfo& info);
 
+// TODO: Only used for TRT and CUDA EP currently, need to add more identifiers to use it across all EPs
+class AllocatorManager {
```

Comment:
Both CUDA and TRT are EPs that work on the same device. Hence they should ideally be given access to a device allocator registry they can access to fetch an allocator as per the device they're working on. This class seems to be in that direction. We used to have a class called DeviceAllocatorRegistry in this file. Not sure why it was [removed](https://github.com/microsoft/onnxruntime/pull/2451).